### PR TITLE
Cookie control API

### PR DESCRIPTION
**Pull request by bararchy**

### CODE REVIEW

**Comment by ysbaddaden** [comment]
Hum, I get that we need a domain to set cookies, but that's weird to have to set the URL for a cookie to be set. Isn't `session.url=` loading the page in the browser already?

**Comment by bararchy** [comment]
@ysbaddaden It is, it feels really weird, but without it the selenium session crashes :\

**Comment by ysbaddaden** [comment]
Looks like this is "by definition", there must be a "browsing context" for the domain we're trying to add the cookie for:
- https://w3c.github.io/webdriver/webdriver-spec.html#dfn-adding-a-cookie

It would be fantastic to have a documentation note that explains this: you must first navigate to the URL you want to list/add/delete cookies for.

While we're on this, webdriver allows to list, get, delete and clear cookies, what about adding methods for these?

**Comment by ysbaddaden** [comment]
Oh, and a test would be fantastic :)

**Comment by bararchy** [comment]
@ysbaddaden Sorry for the delay, I'll try to tackle this the coming week.
I'll at least add the test so it can be merged, and will slowly add the other functionality

**Comment by bararchy** [comment]
@ysbaddaden Added get_cookie and tests.

File: src/webdriver/session.cr [hunk 1]
```diff
@@ -77,21 +77,38 @@ module Selenium
...
     def frame(identifier)
-      post("/frame", { id: identifier })
+      post("/frame", {id: identifier})
     end
 
     def parent_frame
       post("/frame/parent")
     end
 
+    def set_cookie(value, domain, name, path = "/", http_only : Bool = false, secure : Bool = false)
```
Comment (eliasjpr) [inline src/webdriver/session.cr]: why not name the setter `cookie=` and getter `cookie`?

**Comment by ysbaddaden** [comment]
Ideally I'd love a `cookies` value object with a few methods, similar to the following, so we'd have a nice DSL:
- `session.cookies.get("session_id")`
-  `session.cookies.set("username", "julien")`
- `session.cookies.delete("session")`
- `session.cookies.clear`
- ...

```crystal
class Session
  def cookies
    Cookies.new(self)
  end

  struct Cookies
    def initialize(@session : Session)
    end

    def clear
      session.delete("/cookie")
    end

    def delete(name)
      session.delete("/cookie/#{name}")
    end

    def each
      JSON.parse(session.get("/cookie")).each do |json|
        yield json.as_h["cookie"]
      end
    end

    def get(name)
      JSON.parse(session.get("/cookie/#{name}")).as_h["cookie"]
    end

    def set(name, value, domain = nil, path = "/", http_only = false, secure = false)
      cookie = {
          "name" => name,
          "value" => value,
          "path" => path,
          "http_only" => http_only,
          "secure" => secure
      }
      cookie["domain"] = domain if domain
      session.post("/cookie", { cookie: cookie })
    end
  end
end
```

**Comment by bararchy** [comment]
@ysbaddaden it seems you have quite a full solution in mind XD
Maybe I better close the current PR and let you add yours ?

**Comment by ysbaddaden** [comment]
You're right, I went a little overboard. I wrote some tests, changed a few things and pushed #19

**Pull request closed without merging**